server/routes/bookVisit/checkVisitDetailsController.ts

```diff
@@ -47,12 +49,26 @@ export default class CheckVisitDetailsController {
         delete req.session.bookingJourney
         return res.redirect(paths.BOOK_VISIT.BOOKED)
       } catch (error) {
-        // HTTP 400 Bad Request is the response when a session is no longer available
-        if (error.status === 400) {
+        if (error.status === 422) {
+          const validationErrors =
+            (error as SanitisedError<ApplicationValidationErrorResponse>)?.data?.validationErrors ?? []
+
+          if (
+            validationErrors.includes('APPLICATION_INVALID_PRISONER_NOT_FOUND') ||
+            validationErrors.includes('APPLICATION_INVALID_PRISON_PRISONER_MISMATCH')
+          ) {
+            return next(error)
+          }
+
+          if (validationErrors.includes('APPLICATION_INVALID_NO_VO_BALANCE')) {
+            return res.redirect(paths.BOOK_VISIT.CANNOT_BOOK)
+          }
+
```

Comment:
It's likely that these checks and new error code responses will be added to the create/change application in the future. When they are, these conditions can be moved to a separate helper function to avoid duplication.